lib/puppet/pops/loader/loader.rb

```diff
@@ -40,13 +40,18 @@ def initialize(loader_name)
   # found values for which the given block returns true. All found entries will be returned if no
   # block is given.
   #
+  # Errors that occur function discovery will either be logged as warnings or collected by the optional
```

Comment:
This is a significant change in default behavior, but seems reasonable. I'm not sure what use-cases this is targeting, but at least some require continuing beyond failure so it makes sense to provide a way to do that as well as a way to handle actual errors.

Reply:
Will this change behavior when you try to reference a module in code and there are multiple versions on the module path?

Reply:
The only change in default behavior is that load errors during discovery (i.e. listing of loaded entities) are now logged or optionally collected rather than raising an error. I'm not sure why that's considered a significant change so I guess my answer to the question is no. It does not change behavior with respect to multiple versions in the module path.